.pre-commit-config.yaml

```diff
@@ -62,6 +62,20 @@ repos:
         args:
           - --safe
 
+  - repo: https://github.com/pycqa/pydocstyle
+    rev: 6.2.2 # 6.2.3 is slightly broken
+    hooks:
+      - id: pydocstyle
+        args:
+          - --convention=pep257
+          # Additional ignore reasons:
+          # D1xx: we do not want to force contributors to write redundant or useless docstrings
+          # D202: additional whitespace helps with readability
+          # D211: same as D202
+          # See the following documentation for what each rule does:
+          # https://www.pydocstyle.org/en/6.2.3/error_codes.html#error-codes
+          - --add-ignore=D1,D202,D211
```

Comment:
The API is failing on a number of D205, D400, D401, D402 checks as well. This PR will need significant code changes to pass CI.